### ISSUE TITLE
Explorer cannot connect to BigQuery

### ISSUE THREAD

**Issue by quackemi-inmar** [bstruct/vscode-bigquery#63]
The explorer tab does not list any items.  If I attempt to refresh it, I receive the error: 'Unable to detect a Project Id in the current environment.'.  

The authentication tab shows an active account.  I am using a user login via the gcloud CLI, not a service account key.  When I make the attempt to refresh the explorer tab I see the following command in the terminal:

gcloud config get-value project

Which appears to return the project that I have configured as a default in the CLI.

Info on my VS Code setup:
Version: 1.78.2 (user setup)
Commit: b3e4e68a0bc097f0ae7907b217c1119af9e03435
Date: 2023-05-10T14:39:26.248Z
Electron: 22.5.2
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.19045
Sandboxed: No

I am running v0.1.22 of Bigquery Data View.

Please let me know if there is any other info that would be helpful.  Thank you.

**Comment by DamiaoCastro** [comment]
Hi @quackemi-inmar . Sorry to hear that you're having some issues connecting. Unforntunatly, that specific problem in more on the `gcloud` command line than in the VS Code extension.
I will try to come up with alternatives for the API's that the extension interacts to obtain the list. 
In the mean time, can I ask you to re-authenticate, but with 1 extra step, because you're using windows:
Please go to the folder `C:\Users\%username%\AppData\Roaming\gcloud` and delete the file `application_default_credentials.json`. After that, you can go to the extension and revoke the current credentials and ask for new one. The file mentioned is where gcloud caches the information of the login, unfortunately I need to do this operation on my work laptop everytime that I change credentials. Didn't find yet a good way to resolve this in windows. On my dev linux machine I never have this issue.
Please let me know

**Comment by DamiaoCastro** [comment]
In the `terminals` tab, the extension created a read-only sub-tab called `gcloud authentication`
There you can see all the commands that the extension runs and it's outcome.
![image](https://github.com/bstruct/vscode-bigquery/assets/16340298/eb913067-0c68-4a43-a826-6460b53546b3)

the project list comes from the command `gcloud config get-value project` as you can see in the image above.
The reason that I suggested that you delete your credentials and re-login, it's because the extension tries to mark a project as default if there's none in that situation.
probably there's 2 options that you can try.
run `gcloud init` and correct the configuration.
but to debug a bit better, you can run the command `gcloud config set project PROJECT_ID` to confirm that your credentials can actually access the project.
If it possible that you were given access to a dataset or a table, not to a project. The extension allows you to configure projects and dataset's that you were given fine-tuned access:
![image](https://github.com/bstruct/vscode-bigquery/assets/16340298/5aa974fe-7c41-4a40-8ff0-8548a060d9b0)
when you click `Edit in settings.json` you can do something like this:
![image](https://github.com/bstruct/vscode-bigquery/assets/16340298/8c0a3056-cfbd-464c-88e3-97b79fff88c8)

then, if you have permissions to any dataset in that project, you will be able to see it.
Nevertheless, have a default project will be necessary for you to run queries. There always must be a project that the cost of running your queries is associated to.
Hope that any of this information helps.

**Comment by quackemi-inmar** [comment]
Hello.  Apologies for the delay in responding - I have been on vacation.  Adding the project name to the `settings.json` file allowed it to attempt to load the project.  At that point I started getting the error: `Caller does not have required permission to use project xxx. Grant the caller the roles/serviceusage.serviceUsageConsumer role, or a custom role with the serviceusage.services.use permission, by visiting https://console.developers.google.com/iam-admin/iam/project?project=xxx and then retry. Propagation of the new permission may take a few minutes.`

I got the `serviceusage.services.use` permission added to my account and am now able to list out the items in the project with the explorer.  You might want to add a note about that permission in the install, and/or make the error accessible without changing the `settings.json` file.  

Thank you for your help!
